stl/inc/expected

```diff
@@ -685,13 +700,393 @@ public:
         }
     }
 
-    // [expected.object.eq]
+    template <class _Uty = _Err>
+    _NODISCARD constexpr _Err error_or(_Uty&& _Other) const& noexcept(
+        is_nothrow_copy_constructible_v<_Err>&& is_nothrow_convertible_v<_Uty, _Err>) /* strengthened */ {
+        static_assert(
+            is_copy_constructible_v<_Err>, "is_copy_constructible_v<E> must be true. (N4928 [expected.object.obs]/20)");
+        static_assert(
+            is_convertible_v<_Uty, _Err>, "is_convertible_v<G, E> must be true. (N4928 [expected.object.obs]/20)");
+
+        if (_Has_value) {
+            return _STD forward<_Uty>(_Other);
+        } else {
+            return _Unexpected;
+        }
+    }
+
+    template <class _Uty = _Err>
+    _NODISCARD constexpr _Err error_or(_Uty&& _Other) && noexcept(
+        is_nothrow_move_constructible_v<_Err>&& is_nothrow_convertible_v<_Uty, _Err>) /* strengthened */ {
+        static_assert(
+            is_move_constructible_v<_Err>, "is_move_constructible_v<E> must be true. (N4928 [expected.object.obs]/22)");
+        static_assert(
+            is_convertible_v<_Uty, _Err>, "is_convertible_v<G, E> must be true. (N4928 [expected.object.obs]/22)");
+
+        if (_Has_value) {
+            return _STD forward<_Uty>(_Other);
+        } else {
+            return _STD move(_Unexpected);
+        }
+    }
+
+    // [expected.object.monadic]
+    template <class _Fn>
+        requires is_copy_constructible_v<_Err>
+    constexpr auto and_then(_Fn&& _Func) & {
+        using _Uty = remove_cvref_t<invoke_result_t<_Fn, _Ty&>>;
+
+        static_assert(_Is_specialization_v<_Uty, expected>,
+            "expected<T,E>::and_then(F) requires the return type of F to be a specialization of expected. "
+            "(N4928 [expected.object.monadic]/3).");
+        static_assert(is_same_v<typename _Uty::error_type, _Err>,
+            "expected<T,E>::and_then(F) requires the error type of the return type of F to be E. "
+            "(N4928 [expected.object.monadic]/3)");
+
+        if (_Has_value) {
+            return _STD invoke(_STD forward<_Fn>(_Func), _Value);
+        } else {
+            return _Uty(unexpect, _Unexpected);

```

Comment:
Since we know that `_Uty` is a specialization of `expected`, we can use braces to construct `_Uty`, which would follow our preferred convention. Occurs below, also in `or_else` where `_Uty` is constructed from `in_place`.